federation/executor.go

```diff
@@ -59,67 +80,53 @@ func fetchSchema(ctx context.Context, e ExecutorClient, metadata interface{}) (*
 	})
 }
 
-func NewExecutor(ctx context.Context, executors map[string]ExecutorClient) (*Executor, error) {
-	return NewExecutorWithOptionalArgs(ctx, executors, nil)
+type CustomExecutorArgs struct {
+	schemaSyncer              SchemaSyncer
+	optionalArgs              interface{}
+	schemaSyncInternvalSecond int64
```

Comment:
Maybe move this field to be a function of the SchemaSyncer interface?

Reply:
Hm, since the ticker is in the executor, would it make sense to keep the sync interval here too? It might get confusing if in a custom syncer there's another ticker for when it fetches the schema file from some external source?